CRM/Mailing/BAO/Recipients.php

```diff
@@ -137,4 +137,26 @@ public static function reassign($sourceMailingId, $to) {
     }
   }
 
+  /**
+   * @inheritDoc
+   */
+  public function addSelectWhereClause() {
+    $clauses = array(
+      // Append ACL clauses while fetching recipient count or list when check_permissions is TRUE
+      'contact_id' => array(),
+    );
+
+    // TODO: For now we are using logged ib contact ID as it is not possible to
+    // fetch creator id of the mail  for building ACL clauses
+    $contactID = CRM_Core_Session::getLoggedInContactID();
+    list($aclJoin, $aclWhere) = CRM_ACL_BAO_ACL::buildAcl($contactID);
+
+    if (!CRM_Utils_System::isNull($aclWhere)) {
+      $clauses['contact_id'][] = " IN (SELECT id FROM civicrm_contact contact_a $aclJoin WHERE ( 1 ) $aclWhere ) ";
```

Comment:
I am receiving an error in the recipients preview with this (`Column 'id' in field list is ambiguous`). Changing the SELECT statement to

    (SELECT contact_a.id FROM civicrm_contact contact_a $aclJoin WHERE ( 1 ) $aclWhere )

solves this for me.